tests/e2e/network/network_security_group.go

```diff
@@ -619,6 +640,21 @@ var _ = Describe("Network security group", Label(utils.TestSuiteLabelNSG), func(
 				allowedIPv6Ranges = []string{
 					"2c0f:fe40:8000::/48", "2c0f:feb0::/43",
 				}
+
+				// The overlapping IP ranges will be aggregated after reconciled
```

Comment:
@nilo19 Added some overlapping CIDRs to the existing cases. Let’s see how it goes.